src/Raven.Server/Documents/TcpHandlers/SubscriptionDocumentsFetcher.cs

```diff
@@ -82,12 +83,19 @@ public class SubscriptionDocumentsFetcher
 
             using (_db.Scripts.GetScriptRunner(_patch, true, out var run))
             {
-                foreach (var doc in _db.DocumentsStorage.GetDocumentsFrom(
-                    docsContext,
-                    _collection,
-                    startEtag + 1,
-                    0,
-                    long.MaxValue))
+                IEnumerable<Document> documents = _collection switch
+                {
+                    Constants.Documents.Collections.AllDocumentsCollection => 
```

Comment:
just a side note: if subscription for all docs will be created with filter, then for each document we will execute JS to check if it matches or not. This can cause some performance issues.

Reply:
Pretty much the same rule for all subscriptions.
In practice, you typically have a single large collection, so that doesn't matter

Reply:
I'm just saying that there is a significant difference between doing

```
from Employees
```

and
```
from @all_docs where doc.'@metadata'.'@collection' == 'Employees'
```

when it comes to performance. Not only when we are changing something in 'Employees', but also when changing (maybe more often) documents in e.g. 'Logs' collection.

This needs to be documented well and should not be advised to do so if working on one collection.

Reply:
Sure, that is a good point. We can do `from @all_docs(Users, Items)` or something like that in the next itme